File: 大二下/算法分析和设计/PythonCode/第三章_循环和递归/constitute.py
def constitute():
    global a
    a = []
    global total
    total = 0
    for r in range(0, 100):
        a.append(0)
    n = int(input())
    r = int(input())
    a[0] = r
    comb(n, r)
    if n < r:
        return
    print(total)


def comb(n, r):
    global total
    for i in range(n, r-1, -1):
        a[r] = i
        if r > 1:
            comb(i - 1, r - 1)
        else:
            for j in range(a[0], 0, -1):
                print(a[j], end=" ")
            total += 1
            print(end="\n")


if __name__ == '__main__':
    constitute()

File: 大二下/算法分析和设计/JavaCode/src/第3章_循环和递归/Constitute.java
package 第3章_循环和递归;

import java.util.Scanner;

public class Constitute {
    static int total = 0;

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int r = sc.nextInt();
        constitute(n,r);
    }

    static void constitute(int n,int r){
        if (n < r){
            return;
        }
        int[] a = new int[n];
        a[0] = r;
        comb(n,r,a);
        System.out.println(total);
    }

    static void comb(int n,int r,int[] a){
        for (int i = n; i >= r; i--) {
            a[r] = i;
            if(r > 1){
                comb(i-1,r-1,a);
            }else {
                for (int j = a[0]; j > 0 ; j--) {
                    System.out.print(a[j] + " ");
                }
                total = total + 1;
                System.out.println();
            }
        }
    }
}
